spec/components/notification_component_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+RSpec.describe NotificationComponent, type: :component do
+  let(:content) { 'This is content' }
+  let(:style) { 'notice' }
+  let(:dismiss) { true }
+  let(:background) { false }
+  let(:alert) { false }
+  let!(:inline_component) { render_inline(NotificationComponent.new(content: content, style: style, dismiss: dismiss, background: background, alert: alert)) }
+
+  describe '#content' do
+    context 'when content is a string' do
+      it 'renders content in the body' do
+        expect(inline_component.css('.govuk-notification__body').to_html).to include(content)
+      end
+    end
+
+    context 'when content is a hash' do
+      let(:content) { { title: 'Title', body: 'This is the body' } }
+
+      it 'renders the title' do
+        expect(inline_component.css('.govuk-notification__title').to_html).to include(content[:title])
+      end
+
+      it 'renders the body' do
+        expect(inline_component.css('.govuk-notification__body').to_html).to include(content[:body])
+      end
+    end
+  end
+
+  describe '#dismiss' do
+    context 'when dismiss is true' do
+      it 'renders the dismiss link' do
+        expect(inline_component.css('.dismiss-link').to_html).to include(I18n.t('buttons.dismiss'))
+      end
+    end
+
+    context 'when dismiss is false' do
+      let(:dismiss) { false }
+
+      it 'does not render the dismiss link' do
+        expect(rendered_component).to_not include(I18n.t('buttons.dismiss'))
+      end
+    end
+  end
+
+  describe '#dismiss' do
```

Comment:
Not sure you meant that this describes #dismiss? It looks like it describes style.

Reply:
Yep. Mistake I've now fixed.